cmd/kubectl-gadget/utils/trace.go

```diff
@@ -208,65 +213,236 @@ func GenericTraceCommand(
+			kv := strings.Split(pair, "=")
+			if len(kv) != 2 {
+				return "", errors.New("labels should be a comma-separated list of key-value pairs (key=value[,key=value,...])")
+			}
+			labelsSelector[kv[0]] = kv[1]
+		}
 	}
 
+	trace := &gadgetv1alpha1.Trace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: config.GadgetName + "-",
+			Namespace:    "gadget",
+			Annotations: map[string]string{
+				GADGET_OPERATION: config.Operation,
+			},
+			Labels: map[string]string{
+				TRACE_ID: traceID,
+				/*
+				 * Add all this information here to be able to find the trace thanks
+				 * to them when calling getTraceListFromParameters().
+				 */
+				"GadgetName":    config.GadgetName,
+				"nodeName":      config.NodeName,
+				"namespace":     config.NamespaceName,
+				"podName":       config.PodName,
+				"containerName": config.ContainerName,
+				"outputMode":    config.TraceOutputMode,
+			},
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   config.NodeName,
+			Gadget: config.GadgetName,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     config.NamespaceName,
+				Podname:       config.PodName,
+				ContainerName: config.ContainerName,
+				Labels:        labelsSelector,
+			},
+			RunMode:    "Manual",
+			OutputMode: config.TraceOutputMode,
+		},
+	}
+
+	err := createTraces(trace)
+	if err != nil {
+		return "", err
+	}
+
+	return traceID, nil
+}
+
+// getTraceListFromOptions returns a list of traces corresponding to the given
+// options.
+func getTraceListFromOptions(listTracesOptions metav1.ListOptions) (gadgetv1alpha1.TraceList, error) {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return gadgetv1alpha1.TraceList{}, err
+	}
+
+	var traces gadgetv1alpha1.TraceList
+
+	err = traceRestClient.
+		Get().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Into(&traces)
+	if err != nil {
+		return traces, err
+	}
+
+	return traces, nil
+}
+
+// getTraceListFromID returns an array of pointer to gadgetv1alpha1.Trace
+// corresponding to the given traceID.
+// If no trace corresponds to this ID, error is set.
+func getTraceListFromID(traceID string) (gadgetv1alpha1.TraceList, error) {
 	var listTracesOptions = metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("trace-template-hash=%s", traceID),
+		LabelSelector: fmt.Sprintf("%s=%s", TRACE_ID, traceID),
 		FieldSelector: fields.Everything().String(),
 	}
 
-	// Wait until results are ready and fetch all results
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return traces, fmt.Errorf("Error getting traces from traceID %q: %w", traceID, err)
+	}
+
+	if len(traces.Items) == 0 {
+		return traces, fmt.Errorf("No traces found for traceID %q!", traceID)
+	}
+
+	return traces, nil
+}
+
+// SetTraceOperation set the operation of an existing trace.
+// If trace does not exist or Annotations does not contain GADGET_OPERATION key
+// an error is returned.
```

Comment:
I think this comment needs to be updated. This function does not return an error if the `Annotations` map does not contain the `GADGET_OPERATION` key anymore, correct?